qualang_tools/macros/long_wait.py

```diff
@@ -0,0 +1,46 @@
+from typing import Union
+
+import numpy as np
+from qm.qua import declare, wait, for_
+from qualang_tools.addons.calibration.calibrations import u
+
+MAX_WAIT = 2**31 - 1  # units of clock cycles (4ns)
+
+
+def long_wait(wait_time: Union[float, int], *elements: str, threshold_for_looping: int = u.to_clock_cycles(1e6)):
+    """Perform a QUA wait command for longer than the current maximum wait time, 2^31-1.
+
+    This macro unrolls into QUA for-loop of smaller waits if the maximum wait time is exceeded in `wait_time`.
+
+    Note: `wait_time` must be a python number.
+
+    Args:
+        wait_time (Union[float, int]): Desired wait time in units of clock cycles (4ns).
+        *elements (str): Variable length argument list of elements to wait on.
+        threshold_for_looping: Minimum wait time at which `long_wait` uses a QUA for-loop of waits (default 1ms)
+    """
+    i = declare(int)
```

Comment:
Maybe chose a different name, `i` is often used in QUA loops and I am not sure that it will work if the user does:
```
i = declare(int)
with for_(i, 0, i<10, i+1):
    long_wait() 
```
What do you think?

Reply:
And if it works, maybe it is worth adding a test in case it changes in later versions

Reply:
good idea.. I will change it to something really specific and just leave a comment noting that it shouldn't be something commonly used.

Reply:
Yes perfect!